packages/client/src/components/app/Layout.svelte

```diff
@@ -61,7 +60,8 @@
   })
   setContext("layout", store)
 
-  $: validLinks = getValidLinks(links, $currentRole)
+  $: accessibleRoles = $roleStore
+  $: validLinks = getValidLinks(links)
```

Comment:
We should pass in the value of the role store here as a param, otherwise validLinks won't be reactively recomputed when the value of the role store changes, because it's silently used inside the function.

If we just refactor this to
```js
$: validLinks = getValidLinks(links, $roleStore)
```
Then it will be reactive to both values, which it depends on.